### PR TITLE
Scalacheck generators

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR adds a new section for `Generators`

Please @raulraja @dialelo could you review? Thanks

### CODE REVIEW

File: src/main/scala/scalachecklib/GeneratorsSection.scala [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    check(forAll(posNum[Int])(n => (n > 0) == res1))
+
+    check {
+      forAll(listOfN(10, posNum[Int])) { list =>
+        !list.exists(_ < 0) && list.length == res2
+      }
+    }
+  }
+
+  /** '''Conditional Generators'''
```
Comment (dialelo) [inline src/main/scala/scalachecklib/GeneratorsSection.scala]: I guess these single quotes `'` get rendered into headings, am i right? I believe we've used the equal sign `=` elsewhere for this.

**Comment by dialelo** [comment]
Other than the minor comment LGTM!

File: src/main/scala/scalachecklib/GeneratorsSection.scala (hunk 1 above)
Comment (fedefernandez) [inline src/main/scala/scalachecklib/GeneratorsSection.scala]: It's rendered with bold text, I'll change to headings 👍

Comment (dialelo) [inline src/main/scala/scalachecklib/GeneratorsSection.scala]: awesome!

**Pull request merged**